AutomatedTesting/Gem/PythonTests/editor/EditorScripts/EntityOutliner_EntityOrdering.py

```diff
@@ -0,0 +1,145 @@
+
+    # Get the outliner index for the root prefab container entity
+    def get_prefab_container_index():
+        return entity_outliner_model.index(0, 0)
+
+    # Get the outlienr index for the top level entity of a given name
+    def index_for_name(name):
+        root_index = get_prefab_container_index()
+        for row in range(entity_outliner_model.rowCount(root_index)):
+            row_index = entity_outliner_model.index(row, 0, root_index)
+            if row_index.data() == name:
+                return row_index
+        return None
+
+    # Validate that the outliner top level entity order matches the expected order
+    def verify_entities_sorted(expected_order):
+        actual_order = []
+        root_index = get_prefab_container_index()
+        for row in range(entity_outliner_model.rowCount(root_index)):
+            row_index = entity_outliner_model.index(row, 0, root_index)
+            actual_order.append(row_index.data())
+
+        sorted_correctly = actual_order == expected_order
+        Report.result(Tests.entities_sorted, sorted_correctly)
+        if not sorted_correctly:
+            print(f"Expected entity order: {expected_order}")
+            print(f"Actual entity order: {actual_order}")
+
+    # Creates an entity from the outliner context menu
+    def create_entity():
+        pyside_utils.trigger_context_menu_entry(
+            entity_outliner, "Create entity", index=get_prefab_container_index()
+        )
+        general.idle_wait(0.0)
+
+    # Moves an entity (wrapped by move_entity_before and move_entity_after)
+    def _move_entity(source_name, target_name, move_after=False):
+        source_index = index_for_name(source_name)
+        target_index = index_for_name(target_name)
+
+        target_row = target_index.row()
+        if move_after:
+            target_row += 1
+
+        # Generate MIME data and directly inject it into the model instead of
+        # generating mouse click operations, as it's more reliable and we're
+        # testing the underlying drag & drop logic as opposed to Qt's mouse
+        # handling here
+        mime_data = entity_outliner_model.mimeData([source_index])
+        entity_outliner_model.dropMimeData(
+            mime_data, QtCore.Qt.MoveAction, target_row, 0, target_index.parent()
+        )
+        QtWidgets.QApplication.processEvents()
+
+    # Move an entity before another entity in the order by dragging the source above the target
+    move_entity_before = lambda source_name, target_name: _move_entity(
+        source_name, target_name, move_after=False
+    )
+    # Move an entity after another entity in the order by dragging the source beloew the target
+    move_entity_after = lambda source_name, target_name: _move_entity(
+        source_name, target_name, move_after=True
+    )
+
+    expected_order = []
+
+    # 1) Open the empty Prefab Base level
+    helper.init_idle()
+    helper.open_level("Prefab", "Base")
```

Comment:
This is just a reminder for myself. We might want to use `helper.open_level("", "Base")` in the future.